sql/src/main/java/io/crate/execution/jobs/CumulativePageBucketReceiver.java

```diff
@@ -102,7 +102,10 @@ public CumulativePageBucketReceiver(Logger logger,
             pagingIterator,
             this::fetchMore,
             this::allUpstreamsExhausted,
-            () -> processingFuture.complete(null)
+            () -> {
+                releasePageResultListeners();
```

Comment:
I could also initialize the processingFuture like: `new CompletionFuture().handle(... -> releasePageResultListeners()` to enforce *not* missing the release call on all `complete` call-sites. 

What do you think?